elasticsearch/rest/src/main/java/org/apache/metamodel/elasticsearch/rest/ElasticSearchRestDataContext.java

```diff
@@ -133,21 +142,24 @@ public ElasticSearchRestDataContext(final ElasticSearchRestClient client, String
         final List<SimpleTableDef> result = new ArrayList<>();
 
         if (mappings.isEmpty()) {
-            logger.warn("No metadata returned for index name '{}' - no tables will be detected.");
+            logger.warn("No metadata returned for index name '{}' - no tables will be detected.", indexName);
         } else {
-            for (Entry<String, Object> mapping : mappings) {
-                final String documentType = mapping.getKey();
-
-                @SuppressWarnings("unchecked")
-                Map<String, Object> mappingConfiguration = (Map<String, Object>) mapping.getValue();
+            for (final Entry<String, MappingMetaData> mapping : mappings.entrySet()) {
+                final String tableName = mapping.getValue().type();
+                final Map<String, Object> mappingConfiguration = mapping.getValue().getSourceAsMap();
+                
                 @SuppressWarnings("unchecked")
-                Map<String, Object> properties = (Map<String, Object>) mappingConfiguration.get("properties");
-
-                try {
-                    final SimpleTableDef table = detectTable(properties, documentType);
-                    result.add(table);
-                } catch (Exception e) {
-                    logger.error("Unexpected error during detectTable for document type '{}'", documentType, e);
+                final Map<String, Object> properties = (Map<String, Object>) mappingConfiguration.get(ElasticSearchMetaData.PROPERTIES_KEY);
+
+                if (properties != null) {
+                    try {
+                        final SimpleTableDef table = detectTable(properties, tableName);
+                        result.add(table);
+                    } catch (Exception e) {
+                        logger
+                                .error("Unexpected error during detectTable for document mapping type '{}'", tableName,
```

Comment:
Just cosmetic, but this looks rather odd. Is this really how it's supposed to be formatted?

Reply:
On one line it's just over 120 characters, so my code formatter breaks it up into this, but it does look better on one line (and it's still under 130 characters on one line, so that's acceptable, so I'll manually put it on one line and make sure I don't format it).